system/services/assetManager/ImageMagickWrapper.cfc

```diff
@@ -0,0 +1,173 @@
+component output="false" displayname="ImageMagick"  {
+
+	public any function init(
+		 required string  path
+		, 		  numeric timeout = 30
+	) {
+		_setExecutablePath( arguments.path );
+		_setTimeout( arguments.timeout );
+		return this;
+	}
+
+	public binary function resize(
+		  required binary  asset
+		,          numeric width               = 0
+		,          numeric height              = 0
+		,          string  quality             = "highPerformance"
+		,          boolean maintainAspectRatio = false
+		,          struct  assetProperty       =  {}
+	) output=false {
+		var image              = "";
+		var currentImageInfo   = {};
```

Comment:
Is this `assetProperty` argument really required? Is it there just to save the overhead of reading width / height from image file? 

If so, it should probably be called `assetProperties` and the `NativeImageManipulationService` should also use it.